syslog/syslog.go

```diff
@@ -143,10 +143,13 @@ func (l *Logger) writePacket(p Packet) {
 			l.connect()
 		}
 
+		deadline := time.Now().Add(time.Duration(30) * time.Second)
 		switch l.conn.netConn.(type) {
 		case *net.TCPConn, *tls.Conn:
+			l.conn.netConn.SetWriteDeadline(deadline)
 			_, err = io.WriteString(l.conn.netConn, p.Generate(0)+"\n")
 		case *net.UDPConn:
+			l.conn.netConn.SetWriteDeadline(deadline)
```

Comment:
You're not going to ever get into a situation (beyond extreme swapping) where it would take even a second for the kernel to decide to either drop or send a UDP packet.

Having a 30 second timeout on a UDP send just isn't going to actually be something you'll run into in real world situations.